js-envs/src/main/scala/org/scalajs/jsenv/Input.scala

```diff
@@ -29,6 +29,16 @@ object Input {
   /** All files are to be loaded as scripts into the global scope in the order given. */
   final case class ScriptsToLoad(scripts: List[VirtualBinaryFile]) extends Input
 
+  /** All files are to be loaded as ES modules, in the given order.
+   *
+   *  Some environments may not be able to execute several ES modules in a
+   *  deterministic order. If that is the case, they must reject an
+   *  `ESModulesToLoad` input if the `modules` argument has more than one
+   *  element.
+   */
+  final case class ESModulesToLoad(modules: List[VirtualBinaryFile])
+      extends Input
```

Comment:
Is this really a reasonable loading strategy to add? It seems maybe we should just allow loading a single module and force the callers to make things load hierarchically if necessary?

We shouldn't take additions here lightly as we push the burden of supporting them on all JSEnvs. I'm still wondering the same about `CommonJSModulesToLoad` by the way, but here it is more flagrant since generating a simple `import `chain is not even enough.

Reply:
If you put several `<script type="module">` in a .html page, they will be loaded in sequence, as specified by this strategy. Therefore, this strategy is a good proxy for how front-end applications will behave. Also, for the same reason, it's going to be trivial to implement on all JSEnvs that provide a browser-like environment (including Selenium and jsdom; PhantomJS will probably never support modules at all).

If we *don't* support this, then users will be burdened if they need to replicate in their testing environment the loading sequence of several modules that they have in their web page. When using `NodeJSEnv`, they could do what we do currently in `NodeJSEnv`, but if they use a browser-like JSEnv, they'll be hard-pressed to reproduce the setup!